core/Symbols.cc

```diff
@@ -2523,7 +2523,6 @@ namespace {
 void addLocInternal(const core::GlobalState &gs, core::Loc loc, core::Loc mainLoc, InlinedVector<Loc, 2> &locs) {
     for (auto &existing : locs) {
         if (existing.file() == loc.file()) {
-            existing = loc;
```

Comment:
This is really the crux here. In order for this to work, we have to always assign the *first* loc in a file, not the last one. I think this might break other stuff.